articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: build-an-outlook-clone-using-react-hooks
+title: React Hooks and Building Outlook Clone
+description: Complete guide where we discuss react hooks and have written optimized code following up on our earlier topic of how to build an outlook clone using React.
+author: lalithnaryan-c
+date: 2020-07-23T00:00:00-07:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/build-an-outlook-clone-using-react-hooks/hero.jpg
+    alt: outlook clone example image
+---
+Welcome to Hook part of the React series. This time we cover implementing the same outlook clone that we built earlier. But this time, we use React Hooks. The earlier codebase was repetitive and there is a scope for optimization of the entire project. To sum up the earlier attempt at building an outlook clone, we made API calls and displayed emails according to the filter requested. We also implemented local storage.
+
+<!--more-->
+
+Let's think about this for a minute. When we implemented local storage, was it a good idea to store the entire state? This is a factor that needs some consideration of the user interface. Upon reloading, we can simply show the user the set of reading and favorite mails and refresh the layout to the default option. This is the ideal approach to building projects. First, we build them and then work with various optimization in terms of performance, user experience, and code quality.
+
+React class-based components have a bad outlook towards code quality. There is a lot of repetition. DRY, an acronym that stands for Don't Repeat Yourself, effectively speaks for itself. Therefore, React Hooks is a layer on top of the existing library. Let's understand what hooks are and how they work in the next section.
+
+### React Hooks: Why?
+
+In the earlier article, we were constrained to use state and other React features inside the class. We defined the class as
+
+```jsx
+import React from 'react';
+
+class Example extends React.Component{
+    //sample component code here
+}
+```
+The above definition declares that *Example* class is a child component of *React.Component* class.
```

Comment:
This section does not seem complete, more is needed here to understand what point you are trying to make. If the following section is meant to clarify the point you are trying to make then you should make the two sections flow together better.